Feat: post 페이지 edit 상태 구현

ckswnskfk opened the pull request:
## 어떤 변경인지 
- [x] Feat: 기능 변경, 기능 추가
- [ ] Docs: 문서 작업
- [ ] Refactor: 기능 변경 없이 코드 수정
- [ ] Style: 디자인
- [ ] Fix: 버그 수정
- [ ] Test: 테스트 코드 작성

## 설명

post 페이지의 edit 상태를 구현했습니다. 인증, 인가에 따른 상태 변경이 아니라 경로에 따른 상태 변화라 router에 페이지를 추가했습니다.
그런데 막상 사용해보니 삭제 전 확인하는 흐름이 있어야 할 것 같네요.. 만약 모달로 구현하려면 모달을 대대적으로 고쳐야 할 것 같습니다.

반응형은 추후 반영하겠습니다.

### 이슈 번호
https://github.com/Codeit-Rolling-11-Letsgo/Rolling/issues/59

### 주요 변경 사항 
1. edit 페이지 추가
2. useLocation 으로 edit url 체크 및 상태 관리
3. edit 상태에 따른 삭제 버튼 표출
4. 메시지 삭제
5. 삭제 이후 reload
6. 롤링페이퍼(recipient) 삭제
7. 삭제 이후 /list로 리다이렉트
8. 문자열로 전달 받은 html 코드 표현
9. 사용하지 않는 RecipientInfoContext 삭제

### 어떻게 동작하는지

### To Reviewers

ckswnskfk committed:
Feat: post 페이지 edit 상태 구현

ckswnskfk committed:
Feat: post 페이지 edit 상태 구현

ckswnskfk committed:
Fix: 삭제 이후 reload 오류 수정

ckswnskfk committed:
Fix: 삭제 이후 reload 오류 수정

dkile committed:
Feat: 홈페이지 모바일, 태블릿 디자인 구현 (#63)

* Feat: labtop, tablet, mobile breakpoint 변수, mixin 지정

* Feat: 홈페이지 tablet 디자인 구현

* Feat: 홈페이지 mobile 화면 구현

dkile committed:
Feat: 리스트 페이지 모바일, 태블릿 디자인 구현 (#64)

* Feat: touch가 가능한 기기에서는 drag로 캐러셀 이동

* Feat: 리스트 페이지 반응형 디자인 구현

* Style: desktop 화면에서 버튼 위치 조정

* Fix: 리스트 페이지 모바일 레이아웃 마진 수정

mizign committed:
feat: 이모지바 레이아웃 완성 (#60)

* feat: 카카오톡 공유기능 구현 및 레이아웃 완성

* feat: postPage 삭제

* feat: 카카오톡 공유시 페이지 정보 받아오게 구현, 이모지바 레이아웃 완성

* style: borderLine 추가, 버튼 호버 효과 추가

* feat: useLoader 사용해서 recipientId, senderList 등 헤더 서비스에 적용

* review: 불필요한 state삭제, reactionList.length로 대체

* review: 아바타--텍스트간 간격 넓힘

* review: 몇 명 더 있는지 보여주는 아바타 border 제작, 간격 조정

* review: gnb 구분선 적용 태그 수정

* fix: 모바일에서 이모티콘 추가 드롭다운 잘리는 현상 해결

---------

Co-authored-by: So-Yeon Ahn <sozygn@gmail.com>

ckswnskfk commented on src/apis/post/postAPI.js:
```diff
@@ -32,3 +36,19 @@ export const getRecipientInfo = async (recipientId) => {
+
+/**
+ *
+ * @param {number} messageId
+ */
+export const deleteMessage = async (messageId) => {
+	await fetcher.delete(`messages/${messageId}/`).json();
+};
+
+/**
+ *
+ * @param {number} recipientId
+ */
+export const deleteRecipients = async (recipientId) => {
+	await fetcher.delete(`recipients/${recipientId}/`);
+};
```
성공이나 실패 여부에 따라 boolean 타입을 반환해서 다음 동작을 제어하고 싶은데, 이런 요청은 아무런 반환을 해주지 않는 것 같더라구요. 어떻게 성공, 실패를 판단할까요?

dkile replied:
try catch로 감싸는 방법이 좋을 것 같네요

dkile replied:
성공 실패 여부를 모달로 보여주고 싶으신 건가요

ckswnskfk replied:
지금 생각해 보니 toast가 좋을 것 같네요! toast를 사용할 수 있는 hook?도 있으면 좋겠네요ㅎㅎ

ckswnskfk commented on src/pages/post/PostIdPage.jsx:
이 파일 코드가 아주 더러운데 어떻게 관심사의 분리? 리팩터링을 할 수 있을까요 ?.?

dkile committed:
Fix: padding_safe_zone 이름 수정 (#70)

dkile committed:
Fix: 레이아웃 조정 (#74)

* Fix: 레이아웃 조정

* Fix: 중간 정렬 수정

mizign committed:
fix: 다른 드롭다운 열면 열려있던 드롭다운 닫히도록 개선 (#72)

* fix: outside 감지 ref 태그 변경

---------

Co-authored-by: So-Yeon Ahn <sozygn@gmail.com>

dkile commented on src/pages/post/PostIdPage.jsx:
```diff
@@ -1,26 +1,38 @@
+	const { isModalOpen, openModal } = useModalContext();
 
-	const [currentMessageList, setCurrentMessageList] = useState(messageList);
+	const [isEdit, setIsEdit] = useState(false);
```
isEdit은 useLocation 내부 상태에 종속된 친구이기 때문에 변수로 작성해도 충분할 것 같아요.
```
const isEdit = pathname.includes('edit');
```

dkile commented on src/pages/post/PostIdPage.jsx:
```diff
@@ -1,26 +1,38 @@
+	);
 	const [isLoading, setIsLoading] = useState(false);
 
+	useEffect(() => {
```
isEdit state가 없어지면 useEffect도 없어져도 될 것 같아요.

dkile commented on src/pages/post/PostIdPage.jsx:
```diff
@@ -1,26 +1,38 @@
+
 	const fetchMoreMessageList = async () => {
-		if (messageListInfo.count > currentMessageList.length) {
+		if (currentMessageListInfo.count > currentMessageList.length) {
```
어떤 걸 검증하는지 제가 좀 헷갈리는데 설명해주실 수 있을까요? 메세지를 삭제한 경우에만 추가로 불러오는 건가요?

ckswnskfk replied:
스크롤 시 필요하다고 생각했습니다. 검증 로직이 없으면 요청을 계속해서 보내게 되네요.

dkile commented on src/pages/post/PostIdPage.jsx:
```diff
@@ -37,9 +50,25 @@ export default function PostIdPage() {
+		setIsLoading(false);
+	};
+
+	const handleClickDelete = async () => {
```
여기도 try catch를 적용해서 성공한 경우에는 list 페이지로 이동하고 실패한 경우에는 alert나 모달을 띄워도 좋을 것 같네요.

dkile commented on src/pages/post/PostIdPage.module.scss:
```diff
@@ -26,18 +26,38 @@
+	background: $color_purple_600;
+	color: $color_white;
+	cursor: pointer;
+	@include typo_16_regular;
```
mixin은 보통 sass에서 클래스 내 최상단에 작성합니다.

dkile committed:
Fix: 홈페이지, 리스트 페이지 이슈 수정 (#76)

* Fix: service intro 이미지 jpg에서 png로 수정

* Fix: cta active 텍스트 컬러 수정

mizign committed:
feat: postId 페이지 애니메이션 적용 (#79)

* refator: 이모지 뱃지 렌더링 방식 map으로 변경, 공통 컴포넌트 폴더의 이모지 뱃지 사용

* feat: framer motion 애니메이션 함수 제작

* feat: '+'버튼 카드에 호버시 아이콘 90도 회전 애니메이션 적용

* feat: 모달창 팝오버 애니메이션 적용

* feat: 공유버튼 및 토스트 팝업 애니메이션 적용

* feat: 이모지 콜렉션/추가 버튼 팝업 애니메이션 적용

* feat: 카드 호버시 1.02배 확대 애니메이션 적용

---------

Co-authored-by: So-Yeon Ahn <sozygn@gmail.com>

ckswnskfk committed:
Refactor: 리뷰 사항 반영

api try..catch로 감싸 성공, 실패 여부 반환
api 호출하는 곳에서 성공할 때만 다음 작업 수행
isEdit 값은 state 대신 변수로 받음
mixin 클래스 내 상단으로 이동

ckswnskfk committed:
Refactor: 리뷰 사항 반영

api try..catch로 감싸 성공, 실패 여부 반환
api 호출하는 곳에서 성공할 때만 다음 작업 수행
isEdit 값은 state 대신 변수로 받음
mixin 클래스 내 상단으로 이동

ckswnskfk committed:
fix: conflict 해결하다 잘못 들어온 코드 수정

ckswnskfk committed:
Merge branch 'dev' into feature/ckswnskfk/59
